barcodes_generator_product/views/view_product_product.xml

```diff
@@ -11,6 +11,44 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
     <record id="view_product_product_form" model="ir.ui.view">
         <field name="model">product.product</field>
         <field name="inherit_id" ref="product.product_normal_form_view"/>
+        <field name="arch" type="xml">
+            <field name="barcode" position="before">
+                <field name="barcode_rule_id"
+                    domain="[('generate_model', '=', 'product.product')]"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                    groups="barcodes_generator_abstract.generate_barcode" colspan="2"/>
+                <field name="generate_type" invisible="1" />
+                <field name="barcode_base" attrs="{
+                    'invisible': ['|',('barcode_rule_id', '=', False),('product_variant_count', '&gt;', 1)],
+                    'readonly': [('generate_type', '!=', 'manual')]}"
+                    groups="barcodes_generator_abstract.generate_barcode" colspan="2"/>
+                <button name="generate_base" type="object" string="Generate Base (Using Sequence)"
+                        attrs="{'invisible': ['|','|',
+                    ('generate_type', '!=', 'sequence'),
+                    ('barcode_base', '!=', 0),
+                    ('product_variant_count', '&gt;', 1)
```

Comment:
Why to hide it in this case? Isn't it the opposite to what you try to achieve?

Reply:
Hi @chienandalu 
You are very welcome ;)
Actually it's not in the variants form but in the product_normal_form_view that we hide the barcode variants exist.
Regards

Reply:
Well, that's not true when you go to the variant from Catalog > Product Variants :)

Reply:
It's another view?!
I can't find the menu catalog > Products Variants !

Reply:
Sorry. Master Data > Product Variants :)

Reply:
Sorry but I can't find anything like "master data"
It's a menu right? From witch app do I access it?

Reply:
Inventory > Master Data > Product Variants

You can test in this PR runbot

Reply:
I understand now what you meant from the beginning :)
I've committed a fix
thank you for your support